### PR TITLE
feat: add highlighted market on home page

### PR DESCRIPTION
**Pull request by berteotti**
image

### CODE REVIEW

**Comment by Diogomartf** [comment]
> [!TIP]
> Let's improve the "dots", make it more smooth (smaller, better colors)
> Also would be great if we could add some smooth arrows on desktop, to make it easier to swipe. 

image

**Comment by Diogomartf** [comment]
We should only show markets that are open, if none of the highlighted are open, we should not show highlights.

I think we should only highlight markets that have more that 7 days remaining.

File: app/page.tsx [hunk 1]
```diff
@@ -425,3 +435,71 @@ export default function HomePage() {
     </div>
   );
 }
+
+const MarketHighlight = () => {
+  const marketIdList = [
+    '0x8943a769cb0503f7bc4e92a777fcd1aa9decfc33',
+    '0x0539590c0cf0d929e3f40b290fda04b9b4a8cf68',
+  ];
```
Comment (Diogomartf) [inline app/page.tsx:443]: we should have a place for this outside, like a file or something, so we to update it we will only need update that file, making updates on highlights very clear.

File: app/page.tsx [hunk 2]
```diff
@@ -425,3 +435,71 @@ export default function HomePage() {
     </div>
   );
 }
+
+const MarketHighlight = () => {
+  const marketIdList = [
```
Comment (Diogomartf) [inline app/page.tsx:440]:
```suggestion
  const highlightedMarketIdList = [
```

File: app/page.tsx [hunk 3]
```diff
@@ -425,3 +435,71 @@ export default function HomePage() {
...
+                  {/* <p className="text-[44px] font-semibold leading-[56px]"> */}
+                  <p className="text-xl font-semibold">{market.title}</p>
+                </div>
+                <OutcomeBar market={market} />
+                <p className="text-base font-semibold text-text-med-em">
+                  ${formatValueWithFixedDecimals(market.usdVolume, 2)} Vol
+                </p>
+              </div>
+              <Image
+                className="h-full rounded-e-20"
```
Comment (ElRodrigote) [inline app/page.tsx:495]: @see 
https://nextjs.org/docs/pages/building-your-application/optimizing/images#priority

```suggestion
                priority
                className="h-full rounded-e-20"
```

**Comment by Diogomartf** [comment]
we should have a default image in case there in no highlighted image or the market doesn't have one, so we can display a pattern or something for now.

I used "Lisbon" svg from https://heropatterns.com/ for an example but Akash can do a better one.

image

File: app/components/Carousel.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
+'use client';
+
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (berteotti) [reply app/components/Carousel.tsx:3]: We do cause we are using `React` below

File: app/page.tsx (hunk 3 above)
Comment (berteotti) [reply app/page.tsx:495]: Changed to a div with a background which handles mage fallback better than `Image` component

File: app/components/Carousel.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+};
+
+type CarouselContextProps = {
+  carouselRef: ReturnType<typeof useEmblaCarousel>[0];
+  api: ReturnType<typeof useEmblaCarousel>[1];
+  scrollPrev: () => void;
+  scrollNext: () => void;
+  canScrollPrev: boolean;
+  canScrollNext: boolean;
+} & CarouselProps;
```
Comment (Diogomartf) [inline app/components/Carousel.tsx:27]: doesn't the lib provide types ?

File: app/page.tsx [hunk 6]
```diff
@@ -525,3 +544,83 @@ const LoadingMarketCategories = () => (
...
+          className="h-44 w-full rounded-e-0 rounded-t-20 md:h-full md:w-1/2 md:rounded-e-20 md:rounded-s-0 2xl:w-2/5"
+          src={image}
+          alt="Market highlight"
+          style={{ objectFit: 'cover', objectPosition: 'top' }}
+          onError={() => setImage(defaultHighlightImage)}
+        />
+      </Link>
+    </CarouselItem>
+  );
+};
```
Comment (Diogomartf) [inline app/page.tsx:626]: I would extract these component to it's own file

File: market-highlight.config.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+// Image should be stored inside public/assets/highlights
+// import kamalaTrump from '@/public/assets/highlights/kamala-trump.png';
+//
+// ['0x0539590c0cf0d929e3f40b290fda04b9b4a8cf68']: { image: kamalaTrump }
+
+import kamalaTrump from '@/public/assets/highlights/kamala-trump.png';
+import { StaticImageData } from 'next/image';
+
+const highlightedMarketsList: Record<string, { image: StaticImageData }> = {
+  ['0x8943a769cb0503f7bc4e92a777fcd1aa9decfc33']: {
```
Comment (Diogomartf) [inline market-highlight.config.ts:12]: why the array here?

File: package-lock.json
Comment (Diogomartf) [inline package-lock.json:1]: remove package-lock

File: market-highlight.config.ts (hunk 7 above)
Comment (berteotti) [reply market-highlight.config.ts:12]: Removed

File: app/page.tsx (hunk 3 above)
Comment (berteotti) [reply app/page.tsx]: Changed to use Image, applied `prioirty` thanks

File: market-highlight.config.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+// Example market highlight definition
+//
+// Image should be stored inside public/assets/highlights
+// import kamalaTrump from '@/public/assets/highlights/kamala-trump.png';
+//
+// ['0x0539590c0cf0d929e3f40b290fda04b9b4a8cf68']: { image: kamalaTrump }
```
Comment (Diogomartf) [inline market-highlight.config.ts:6]:
```suggestion
// '0x0539590c0cf0d929e3f40b290fda04b9b4a8cf68': { image: kamalaTrump }
```

File: app/MarketHighlight.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  CarouselPrevious,
+  CarouselSelector,
+} from './components/Carousel';
+import Autoplay from 'embla-carousel-autoplay';
+import { highlightedMarketsList } from '@/market-highlight.config';
+import defaultHighlightImage from '@/public/assets/highlights/default.png';
+import Image from 'next/image';
+import { useState } from 'react';
+
+export const MarketHighlight = () => {
```
Comment (Diogomartf) [inline app/MarketHighlight.tsx:22]: The name should be plural as we are highlighting markets
```suggestion
export const MarketsHighlight = () => {
```

**Review by Diogomartf — APPROVED** [review]

**Pull request merged**